File: python_src/bubble_sort.py
def sort_by_bubble_sort(array):
    done = False

    while not done:
        done = True

        for i in range(len(array) - 1):
            if array[i] > array[i + 1]:
                done = False

                array[i], array[i + 1] = array[i + 1], array[i]

File: src/bubble_sort.rs
/// Sorts the given array, in place, by using the Insertion sort algorithm.
pub fn sort_by_bubble_sort(array: &mut [usize]) {
    let mut done = false;

    while !done {
        done = true;

        for index in 0..array.len() - 1 {
            if array[index] > array[index + 1] {
                done = false;

                array.swap(index, index + 1)
            }
        }
    }
}